### PR TITLE
[SYCL-MLIR][NFC]: Move memory access analysis to polygeist

### PR DESCRIPTION
**Pull request by etiotto**
Move the memory access analysis from `mlir-sycl` to `polygeist`.

### CODE REVIEW

File: .github/workflows/sycl_linux_build_and_test.yml [hunk 1]
```diff
@@ -111,6 +111,7 @@ jobs:
       run: cmake --build $GITHUB_WORKSPACE/build --target deploy-sycl-toolchain
     # TODO allow to optionally disable in-tree checks
     - name: check-mlir-sycl
+      shell: bash
```
Comment (whitneywhtsang) [inline .github/workflows/sycl_linux_build_and_test.yml:114]: Why add `shell: bash`? I see that the `sycl_linux_build_and_test.yml` in `sycl` branch https://github.com/intel/llvm/blob/sycl/.github/workflows/sycl_linux_build_and_test.yml doesn't do that.

File: polygeist/lib/Dialect/Polygeist/Analysis/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+add_mlir_dialect_library(MLIRPolygeistAnalysis
+  MemoryAccessAnalysis.cpp
+
+  ADDITIONAL_HEADER_DIRS
+  ${MLIR_MAIN_INCLUDE_DIR}/mlir/Dialect/Polygeist/Analysis
+
+  LINK_LIBS PUBLIC
+  MLIRAnalysis
+  MLIRPolygeistUtils
+)
```
Comment (whitneywhtsang) [inline polygeist/lib/Dialect/Polygeist/Analysis/CMakeLists.txt:10]:
```suggestion
)

```

File: .github/workflows/sycl_linux_build_and_test.yml (hunk 1 above)
Comment (etiotto) [reply .github/workflows/sycl_linux_build_and_test.yml:114]: Because we are already using it for `check-cgeist` at line 119.

Comment (etiotto) [reply .github/workflows/sycl_linux_build_and_test.yml:114]: Is there a reason you prefer to remove the shell ? I can go either way as long as we are consistent.

Comment (whitneywhtsang) [reply .github/workflows/sycl_linux_build_and_test.yml:114]: to be consistent with sycl branch.

Comment (whitneywhtsang) [reply .github/workflows/sycl_linux_build_and_test.yml:114]: Looks like we intentionally added it in https://github.com/intel/llvm/pull/7085, let's keep it then.

Comment (etiotto) [reply .github/workflows/sycl_linux_build_and_test.yml:114]: On further thoughts I am splitting the PR. This one will deal with moving the analysis files. A future one will deal with adding the polygeist unittests to the build scripts.

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**